packit_service/service/api/projects.py

```diff
@@ -0,0 +1,143 @@
+from http import HTTPStatus
+from logging import getLogger
+from flask import make_response
+from json import dumps
+
+try:
+    from flask_restx import Namespace, Resource
+except ModuleNotFoundError:
+    from flask_restplus import Namespace, Resource
+
+from packit_service.service.api.parsers import indices, pagination_arguments
+from packit_service.models import GitProjectModel
+
+logger = getLogger("packit_service")
+
+ns = Namespace("projects", description="Repositories with packit-service enabled.")
+
+
+@ns.route("")
+class ProjectsList(Resource):
+    @ns.expect(pagination_arguments)
+    @ns.response(HTTPStatus.PARTIAL_CONTENT, "Copr builds list follows")
+    def get(self):
+        """List all GitProjects"""
+
+        result = []
+        first, last = indices()
+
+        projects_list = GitProjectModel.get_projects_range(first, last)
+        for project in projects_list:
+            project_info = {
+                "namespace": project.namespace,
+                "repo_name": project.repo_name,
+                "project_url": project.project_url,
+                "num_prs": len(project.pull_requests),
+                "num_branches": len(project.branches),
+                "num_releases": len(project.releases),
+                "num_issues": len(project.issues),
```

Comment:
Do you happen to know what will be the performance of this on the database side?

Reply:
I am thinking if this is anyhow useful. (And those are only prs, issues, releases saved by packit, not all of them..)

Reply:
Its fast on a super weak single core vm with the entire prod db dump you sent me. But its not running any other background tasks unlike the actual deployment which might be processing multiple jobs in the background.

Reply:
> I am thinking if this is anyhow useful.

I thought this will give some quick overview about how active and how often a project uses packit-service in the all projects page.

Reply:
OK, so some kind of statistics?

- I would just make sure it's about packit-served entities, not the upstream ones.
- What about having this only for one-project view. OR, do we want to have the same content?

Reply:
The use of this endpoint is to populate the [projects page](https://dashboard.packit.dev/projects) on the Dashboard, which we planned to be a list of projects and displaying also some high level statistics of those projects. This is where this data should fit, allowing one to have a high level overview of how much and what kind of work Packit was doing on which projects.

From this page one will be able to navigate to the projects individual information page which will display the list of PRs, releases, branches and issues that Packit handled.

> I would just make sure it's about packit-served entities, not the upstream ones.

`project` is of type `GitProjectModel`, so yes, these are Packit-served entities not the upstream ones.

Reply:
Do we need these statistics on the all projects view? If we do, then lets keep this, otherwise we'll remove it. IDK.

Reply:
> Do we need these statistics on the all projects view? If we do, then lets keep this, otherwise we'll remove it. IDK.

I say, yes, we need this information.

Reply:
> From this page one will be able to navigate to the projects individual information page which will display the list of PRs, releases, branches and issues that Packit handled.

That makes sense.

> `project` is of type `GitProjectModel`, so yes, these are Packit-served entities not the upstream ones.

- I know that, but it does not need to be clear for others. (To not need to dig into the code. The database structure can be hidden behind the API. We might want to use API without knowing about the database.)
- I know the API's only user is dashboard now but we can make it in a way that makes sense in general. (Please, don't couple the API and dashboard more than necessary. The API should make sense also alone.)

What about this:
- Use explicit attribute names in the responses and/or
- document the API endpoint (to see it in the main API page; i.e. https://stg.packit.dev/api/), not sure how easy it is to describe results more precisely.

Reply:
Do I understand it correctly that the thing you want to improve is the naming/documentation of the `num_*` fields?

Would rename them to `*_handled` improve the situation? Like: `prs_handled`, `branches_handled`, `releases_handled`, `issues_handled`?

I think documenting responses is possible with Swagger, but we might need to define a schema for that. I would say that's out of scope for this PR.

Reply:
Exactly.

And originally, I was concerned about usefulness of the values what you've explained.

Thanks.